packages/sdk/src/oracle/services/update.ts

```diff
@@ -21,20 +21,26 @@ export class Update {
     const request = params || this.read().inputRequest();
     const chainId = request.chainId;
     const oo = this.read().oracleService(chainId);
-    const fullRequest = await oo.getRequest(
+    // pull in data from contract on chain
+    const contractRequest = await oo.getRequest(
       request.requester,
       request.identifier,
       request.timestamp,
       request.ancillaryData
     );
+    // pull in latest request state
     const state = await oo.getState(request.requester, request.identifier, request.timestamp, request.ancillaryData);
+    // pull in request data generated from events
+    const requestIndexData = this.read().sortedRequestsService().getByRequest(request);
     this.write((write) => {
       // create the erc20 service to handle currency
-      write.services(chainId).erc20s(fullRequest.currency);
+      write.services(chainId).erc20s(contractRequest.currency);
       write
         .chains(chainId)
         .optimisticOracle()
-        .request(request, { ...fullRequest, state });
+        // update request object with all the data we have about it. order is important,
+        // we want to prioritize latest state pulled from contract.
+        .request({ ...requestIndexData, ...request, ...contractRequest, state });
```

Comment:
this function gets called periodically and is responsible for the bulk of updating the active request